app/routes/notes.tsx

```diff
@@ -36,35 +54,55 @@ export default function NotesPage() {
         </Form>
       </header>
 
-      <main className="flex h-full bg-white">
-        <div className="h-full w-80 border-r bg-gray-50">
-          <Link to="new" className="block p-4 text-xl text-blue-500">
-            + New Note
-          </Link>
-
-          <hr />
+      <main className="flex h-full overflow-x-hidden bg-white">
+        <button
+          className="border-r bg-gray-50 p-2 leading-none md:hidden"
+          onClick={toggleMenu}
+        >
+          n<br />o<br />t<br />e<br />s
+        </button>
 
-          {data.noteListItems.length === 0 ? (
-            <p className="p-4">No notes yet</p>
-          ) : (
-            <ol>
-              {data.noteListItems.map((note) => (
-                <li key={note.id}>
-                  <NavLink
-                    className={({ isActive }) =>
-                      `block border-b p-4 text-xl ${isActive ? "bg-white" : ""}`
-                    }
-                    to={note.id}
-                  >
-                    📝 {note.title}
-                  </NavLink>
-                </li>
-              ))}
-            </ol>
+        <div
+          className={cn(
+            "h-full overflow-x-hidden border-r bg-gray-50 md:w-80 md:min-w-[20rem]",
+            "transition-all duration-150 ease-out",
+            {
+              "w-80 min-w-[20rem]": menuOpen,
+              "w-0 min-w-0": !menuOpen,
+            }
           )}
```

Comment:
Do we really need to use `classnames`?

Can't we just do it manually?

```suggestion
          className={`h-full overflow-x-hidden border-r bg-gray-50 transition-all duration-150 ease-out md:w-80 md:min-w-[20rem] ${
            menuOpen ? "w-80 min-w-[20rem]" : "w-0 min-w-0"
          }`}
```

Reply:
We can, but I think most people who are going to keep tailwind around will eventually want to use clsx eventually. It's pretty darn useful.